### ISSUE TITLE
NVDA does not announce the selected values, and it breaks the keyboard navigation.

### ISSUE THREAD

**Issue by UdhayaKumarDuraisamy** [nvaccess/nvda#15916]
I have a custom MultiSelect component built using JavaScript. When selecting values from the component popup, the NVDA screen reader does not announce the selected values. Additionally, after selecting values, the keyboard navigation does not work properly. The keyboard navigation only breaks when using NVDA.

I have tried adding 'aria-activedescendant', 'aria-describedby', and 'aria-labelledby', but none of these resolved the issue.

**Sample**: https://stackblitz.com/edit/2tmhcn?file=index.js

### Steps to reproduce:

- Run the sample mentioned above.
- Focus on the component using Tab key navigation.
- Press the "Alt + Down Arrow" keys to open the popup.
- Use the Down/Up arrows to navigate through the list items in the popup.
- Select a value by pressing the space key.
- After selecting a value, try to move any other list item in the popup.

### Actual behavior:
The NVDA does not announce the selected values properly, and the keyboard navigations are not working properly.

### Expected behavior:
While selecting/deselecting any values the NVDA should announce it and After that keyboard navigation needs to work properly.

### NVDA logs, crash dumps and other attachments:
![135](https://github.com/nvaccess/nvda/assets/98140917/03fd31da-c97d-437a-998f-46887e3787a5)

### System configuration
Processor : 12th Gen Intel(R) Core(TM) i5-1235U   1.30 GHz

#### NVDA installed/portable/running from source:
- Installed from the NVDA wesite https://www.nvaccess.org/download/.
#### NVDA version:
![image](https://github.com/nvaccess/nvda/assets/98140917/6adc8bcc-f0f0-454f-bee2-5b9387d419f5)

#### Windows version:
Edition	: Windows 11 Enterprise
Version	: 22H2

#### Does the issue still occur after restarting your computer?
Yes.

**Comment by CyrilleB79** [comment]
Welcome @UdhayaKumarDuraisamy!

Could you please edit your initial comment, so that it presents the necessary information without use of screen shots? Most of NVDA developers are blind.
Thanks.

**Comment by Adriani90** [comment]
@UdhayaKumarDuraisamy welcome to the NVDA project.
This has the same underlying reason. It happens due to the fact that NVDA provides a virtual navigation mode called browse mode in which the keyboard is overtaken by the screen reader in order to provide a more efficient way of navigating. You can turn browse mode off by pressing nvda key+space bar or by disabling the coresponding setting in NVDA browse mode settings.
Closing as invalid.

Also please don't use only screen shots to describe things, but insert a text showing what is on the screenshot. There are lots of blind developers who cannot investigate because we cannot read what's on the screenshot.

I also encourage you to study the user guide of NVDA, there is everything you need to know about browse mode and focus mode and how NVDA works.

**Comment by UdhayaKumarDuraisamy** [comment]
@Adriani90 , As per the suggestion, I have tried turning off browse mode and the keyboard navigations are working fine. However, it does not read the selected values. Could you please advise what I am doing wrong in my sample? I have added all the necessary attributes.

**Sample**: https://stackblitz.com/edit/2tmhcn-suquy2?file=index.js

**Comment by Adriani90** [comment]
@UdhayaKumarDuraisamy thanks for the info. Indeed there is an issue with these lists. In fact I can hear the values selected and I can navigate the selected values both in Firefox and Chromium browsers, but the focus jumps to a text border or section outside the list when pressing space bar. I will create a new issue for that, but I think it is related to the focus redirection which should point to the combo box which opens the list and not outside of it after pressing space bar on a country.

To hear the selected values:
1. Open NVDA and the test case
2. Use tab or in browse mode the arrow keys or the letter c to navigate to the combo box "choose a country"
3. make sure you are in focus mode, press alt+down arrow to open the list of countries
4. The focus will be placed in the search field. Press down arrow to navigate the list of available countries
5. Press space bar on a country you want to choose
6. The list closes and the focus will move outside of the combo box. Press tab to land on the combo box again and hear the selected values.
7. Press down arrow key to navigate both selected values and available values. Note that selected values are moved to the top of the list, not sure if it is a separate list or the same one.

Issue:
actual behavior: in step 5 the focus is moved outside of the combo box, and you don't hear that the value has been selected.

Expected behavior: 
1. The focus should not move at all after pressing space bar on a country. 
2. NVDA should report "<country> selected" when pressing space bar
3. The list should remain open until you are finished with the selections. The user should be able to close the list by pressing escape.
4. After that the focus should land on the combo box which opens a list and not outside of it.
